src/hooks/use-sync-states-progress-info.ts

```diff
@@ -96,7 +114,7 @@ export function useSyncStatesProgressInfo() {
 				progress: 100,
 				message: __( 'Error pushing changes' ),
 			},
-		} as const;
+		} satisfies PushStateProgressInfoValues;
```

Comment:
Instead of using satisfies, let's use a regular type when declaring the cariable.
```suggestion
		}
```
Something like:

```tsx
	const pullStatesProgressInfo: PullStateProgressInfoValues = useMemo( () => {
```

Reply:
The problem with using a type here in the `useMemo` is that typescript will apply _type widening_, to the return type, meaning that it will be more flexible. So for example:
#### 1. Using a type
```jsx
	const pullStatesProgressInfo = useMemo< PullStateProgressInfoValues >( () => {
```
We could have later on something that would be valid for Typescript:
```jsx
	pullStatesProgressInfo.downloading = {
		key: 'in-progress',
		progress: 100,
		message: __( 'Hello' ),
	};
```
Please note that we are assigning to `key` a different key than `downloading`

#### 2. Using satisfies
![CleanShot 2025-05-15 at 16 03 53@2x](https://github.com/user-attachments/assets/56de7694-4522-43ef-b756-84a380e10706)


There is a much longer explanation in this [chatgpt thread](https://chatgpt.com/share/6825f192-8694-800d-902f-757ccea0eb31) but I think the above is the gist of it. I think this applies here because it is the return of `useMemo`, usually defining and using types is the common approach.

And a [ts playground to try ](https://www.typescriptlang.org/play/?#code/KYDwDg9gTgLgBDAnmYcAKBXAzgCwMowCGMwaUEA5lMFlgJIB2AZhHALxwDeAUAJADWwRAC44AcgDG1YgEsGFAEKEJ-DGDFwAPuLUAbCIQAmcihoDcfMOSo0sohhgC2AI2BQLvR7cIVgorDBQJhYAvhbcoJCwCMiomLgExKTW1LSMLABqhLoYNOxwAErAEtCGADx88fhEJGSUqfTMEADa4oKIGgC6ADSV2NVJdTZpTdwAfOHc3CUMAXCOiFlQAKpYJgAqsXCiVYm1KbbpEFk5eRw8cHBSwLLySipqoheXcO2iktIwJveq6r0vcCs9VsogADP8Xl5aD4-OIcMBdPoxNxLiEIXoDMZ5E8US83jowPojCYxBDLkDhnY4ABGcG4y5QrAw97wjq4kLcDnTCCzeALJarEx4WRYJgyM5cXHXW6KZS-HEA-EfG5fO5ytSk+mAg60MFk+beXwshFI9nowmYkwKvFCd4Y4nyTUAikNUS0-WM5lwoTI1GcuBMr6i8VYdD9PbJYEjTLZXJYKb8wgrNbyTYoAB00tVsoeYHyzyV9qxpghLpBdIZhthYlZyK5ieTQpFYpomc+33VefOuMLFodJdxZd1NIrBuhRrhEDrUxnM6AA)

Reply:
Thanks for sharing more context. It makes sense.
What do you think if we leave also the `as const` and prevent any object manipulation?
Something like:
`} as const satisfies PushStateProgressInfoValues;`

Reply:
Great suggestion, I have added that as part of [332752a](https://github.com/Automattic/studio/pull/1389/commits/332752a4189beedef28add4f03474e2b18b1d64f) and removed two unnecessary exports